store/listenkv/store_test.go

```diff
@@ -149,13 +140,12 @@ func TestListenKVStoreDelete(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		var buf bytes.Buffer
+		listener := types.NewMemoryListener()
 
-		store := newListenKVStore(&buf)
-		buf.Reset()
+		store := newListenKVStore(listener)
 		store.Delete(tc.key)
-		storeKVPair := new(types.StoreKVPair)
-		testMarshaller.UnmarshalLengthPrefixed(buf.Bytes(), storeKVPair)
+		cache := listener.PopStateCache()
+		storeKVPair := cache[len(cache)-1]
```

Comment:
How do you know that PopStateCache returns a slice of length > 0?

Reply:
fixed. added `require.NotEmpty(t, cache)` one line above.